angr/factory.py

```diff
@@ -24,6 +24,9 @@ def __init__(self, project, default_engine=None):
         self.default_engine = default_engine(project)
         self.procedure_engine = ProcedureEngine(project)
 
+        if project.concrete_target:
+            self.concrete_engine = SimEngineConcrete(project)
```

Comment:
can you make it so that the property is present but None even if project doesn't have a concrete target? Inconsistent object shape is a bad idea.